### PR TITLE
feat(runbook): Adding Orca QoS docs

### PR DESCRIPTION
**Pull request by robzienert**

### CODE REVIEW

File: guides/runbooks/orca-quality-of-service.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+layout: single
+title:  "Orca Quality of Service"
+sidebar:
+  nav: guides
+---
+
+{% include toc %}
+
+**EXPERIMENTAL**: This feature is still in an early adoption / experimental phase. 
+While you can use it today (Orca v6.71.0), Netflix is still currently running this in learning mode / judiciously enabling in response to on-call events.
```
Comment (tomaslin) [inline guides/runbooks/orca-quality-of-service.md]: Not sure `judiciously enabling in response to on-call events` is needed

File: guides/runbooks/orca-quality-of-service.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+Spinnaker ships with an optional Quality of Service (QoS) module that can be used to manage the amount of active executions running at any given time.
+By default, this QoS module is disabled, but can be enabled and tuned with a handful of knobs and different strategies.
+Before we dive into the configuration settings, we'll go over how QoS works.
+
+# QoS Internals
```
Comment (tomaslin) [inline guides/runbooks/orca-quality-of-service.md]: Maybe change the heading from Internals to `How does QoS work?` This seems needed for configuration later but the heading Internals make me want to skip this section.

File: guides/runbooks/orca-quality-of-service.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+The QoS system is self-contained within Orca's [orca-qos][module] module and operates only on newly created executions.
```
Comment (tomaslin) [inline guides/runbooks/orca-quality-of-service.md]: are these pipeline executions or does it also include orchestrations, make it more explicit

File: guides/runbooks/orca-quality-of-service.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+When an execution is `BUFFERED`, it will effectively stay in a waiting state until it is unbuffered, which we'll go over later.
+
+`BufferPolicy` functions are pluggable and can contain arbitrary logic. 
+For example, one `BufferPolicy` that ships always enabled is [EnqueueDeckOrchestrationBufferPolicy][deck-buffer-policy], which will always `ENQUEUE` an execution it is an Orchestration and from the UI. 
+This specific policy forces the `ENQUEUE` status, even if other policies call for the execution to be buffered; this is done through a `force` flag that policies can return.
+An example of other pluggable behavior is determining buffering action based on criticality of the execution: At Netflix we have a custom concept of application criticality, so we can buffer low criticality executions to allow capacity for higher criticality executions.
+
+Once an execution is put into a `BUFFERED` state, it will remain in that state until the [ExecutionPromoter][promoter] decides to change it to `NOT_STARTED` and enqueue it for processing.
```
Comment (tomaslin) [inline guides/runbooks/orca-quality-of-service.md]: a diagram of this flow would be useful

File: guides/runbooks/orca-quality-of-service.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+These configurations are not guaranteed to be fully inclusive of all knobs.
+A definitive list is available via the codebase.
+
+* `qos.enabled`: Boolean (default `false`). Global flag controlling whether or not the system is enabled. This flag will not disable the `ExecutionPromoter`.
+* `qos.learningMode.enabled`: Boolean (default `true`). If enabled, executions will always be `ENQUEUED`, but log messages & metrics will be emitted saying what the system would have done. This flag has no effect on `ExecutionPromoter`.
+* `pollers.qos.promoteIntervalMs`: Integer (default `5000`). The time (in milliseconds) that the promotion process will be run.
+
+## BufferPolicy: Naive
+
+The `NaiveBufferPolicy` will always buffer executions when enabled.
```
Comment (tomaslin) [inline guides/runbooks/orca-quality-of-service.md:92]: is there just one bufferpolicy? or are we expecting to add more?

Comment (robzienert) [reply guides/runbooks/orca-quality-of-service.md:92]: Right now there's just one. We'll be adding our own in `-nlfx` for `AppCriticalityBufferPolicy`.

**Comment by tomaslin** [comment]
LGTM

**Comment by robzienert** [comment]
@lwander Could you take a look at the mermaid stuff? It didn't render locally, but it looks the same as other examples; I've included the mermaid lib and see it being pulled in. Not sure if it's something wrong with local previews.

**Comment by tomaslin** [comment]
I couldn't get the diagram to render without removing the two `alt` blocks

```
    alt ENQUEUED
        BufferPolicy->>ExecutionLauncher: Start Execution
    end
    alt BUFFERED
        Note over BufferPolicy: Set BUFFERED status
    end
```

Might be possible that the version used by us ( 6.0.0 ) doesn't support that. 

screenshot 2018-05-22 20 40 54

**Comment by lwander** [comment]
@ttomsu is the mermaid wizard -- maybe he knows?

**Comment by ttomsu** [comment]
I think it needs to be:

```
    alt ENQUEUED
        BufferPolicy->>ExecutionLauncher: Start Execution
    else BUFFERED
        Note over BufferPolicy: Set BUFFERED status
    end
```

**Comment by robzienert** [comment]
Cheers, @ttomsu you're the best.

**Comment by robzienert** [comment]
You other lot are also the best, but Travis is bestest today.

**Pull request merged**